waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -950,67 +960,92 @@ proc appendRLNProof*(rlnPeer: WakuRLNRelay, msg: var WakuMessage,
 proc addAll*(wakuRlnRelay: WakuRLNRelay, list: seq[IDCommitment]): RlnRelayResult[void] =
   # add members to the Merkle tree of the  `rlnInstance`
   ## Returns an error if it cannot add any member to the Merkle tree
-  for i in 0..list.len-1:
-    let member = list[i]
-    let memberAdded = wakuRlnRelay.insertMember(member)
-    if not memberAdded.isOk():
-      return err(memberAdded.error())
+  let membersAdded = wakuRlnRelay.insertMembers(0, list)
+  if not membersAdded.isOk():
+    return err("failed to add members to the Merkle tree")
   return ok()
 
-type GroupUpdateHandler* = proc(pubkey: Uint256, index: Uint256): RlnRelayResult[void] {.gcsafe.}
+type MembershipTuple* = tuple[index: MembershipIndex, idComm: IDCommitment]
```

Comment:
No, imo we shouldn't be creating objects for just 2 fields.

Reply:
I agree, this type has some overlap with other types. In a follow up PR, I will refactor the `RlnMembershipCredentials` type to have the idKey optional. This will allow reuse of the types. wdyt?